group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -6128,11 +6183,11 @@ private CoordinatorResult<LeaveGroupResponseData, CoordinatorRecord> classicGrou
                     // in which case we expect the MemberId to be undefined.
                     if (!UNKNOWN_MEMBER_ID.equals(memberId)) {
                         throwIfInstanceIdIsFenced(member, groupId, memberId, instanceId);
+                        throwIfMemberDoesNotUseClassicProtocol(member);
```

Comment:
This is the fix for https://issues.apache.org/jira/browse/KAFKA-18116. @dongnuo123 What do you think about it?

Reply:
I suppose that we should. However, as it is orthogonal to this PR, I suggest to tackle it separately. Could you please file a jira for it?

Reply:
https://issues.apache.org/jira/browse/KAFKA-18150